docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,140 @@
+        "AHU-TargetTemperature": "28.5",
+        "AHU-TargetBand": "4.0",
+        "AHU-TargetHumidity": {
+        "Accuracy": "0.2-0.3% RH",
+        "Value": 59
+    }
+}
+```
+
+Response messages would be similarly wrapped so as to preserve the correlation id - if nothing else.
+
+``` json
+{
+“correlation-id”: “14a42ea6-c394-41c3-8bcd-a29b9f5e6835”
+“response”:
+    {
+        "apiVersion": "v2",
+        "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+        "statusCode": 0,
+        "message": "string"
+    }
+}
+```
+
+### Configuration
+
+The configuration or the command service must provide the details in order to subscribe to command messages from any 3rd party system and publish to a response topic.  The subscription configuration would be similar to how core data is configured to subscribe to messages from any device service and publish to topics to get messages to application services.  However, the command service will actually need to have two sets of subscription and two sets of publishing topics.
+
+The command service would subscribe to any command topic (“edgex/request/command/#”) to receive requests from the 3rd party system.  It must then relay these messages (via publish) to the device service actuation topics.
+
+As each device service will have a topic assigned to it, the device service name and device name will be added to the destination topic (“edgex/request/actuate/<device-service>/<device-name>”) when command relays the actuation request messages to device services.
+
+The command service must then subscribe to another topic (“edgex/response/actuate/#”) to receive actuation responses from the device services.  It then relays those responses on to the 3rd party apps by publishing those responses to a response topic (“edgex/response/command”).
+
+![image](command-service-topics.png)
+
+``` toml
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+SubscribeResponseActateTopic = “edgex/response/actuate/#”
+PublishActuateTopicPrefix = "edgex/actuate/ " # /<device-service>/<device-name> will be added to this Publish Topic prefix
+SubscribeRequestCommandTopic = "edgex/request/command/#"
+PublishCommandResponseTopic = “edgex/response/command”
+```
+
+Device services must similarly be configured for subscribing to actuation commands and responding to them by publishing to the response topic.
+
+``` toml
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+SubscribeRequestActateTopic = “edgex/request/actuate/<device-service>/#”
+PublishActuateResponseTopic = “edgex/response/actuate/<device-service>/”
+```
+
+Questions:
+-  Would we need multiple 3rd party response topics?  Or would one setup separate command services for each 3rd party?
```

Comment:
Seems to me like it would work without considering what third party requested. Also it seems to me that one command service would suffice to handle all commands for all device services.